framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -587,37 +593,35 @@ def _applyFunctionalConstraints(self, suggested, previous):
   # Utility Methods #
   ###########
   def _temperature(self, fraction):
-    #return max(0.01,min(1,1-fraction))
+    """
+    A utility function to compute the initial temperaturte
+    currently it is just a function of how far in the process are we
+    @ In, fraction, float, the current iteration devided by the iteration limit i.e., $\frac{iter}{Limit}$
+    @ Out, T0, float, initial temperature, i.e., $T0 = max(0.01,1-fraction) $
+    """
     return max(0.01,1-fraction)
 
   def _coolingSchedule(self, iter, T0):
-    """TODO write the math here"""
-    #defaults
-    if T0 == None:
-      T0 = 1e4
-
-    if not self._coolingMethod:
-      self._coolingMethod = 'exponential'
-
-    if not self._coolingParameters:
-      self._coolingParameters['alpha'] = 0.94
-      self._coolingParameters['beta'] = 0.1
-      self._coolingParameters['c'] = 1.0
-      self._coolingParameters['d'] = 1.0
-
+    """
+    A utility function to compute the current cooled state temperature
+    based on the user-selected cooling schedule methodology
+    @ In, iter, int, the iteration number
+    @ In, T0, float, The previous temperature before cooling
+    @ Out, Tk, float, the cooled state temperature i.e., $T^{k} = f(T^0, coolingSchedule);$ where k is the iteration number
```

Comment:
this should be 
```
  @ Out, _coolingSchedule,  the cooled state temperature i.e., $T^{k} = f(T^0, coolingSche
```

I know... it is tedious but unforntelly it is our standard now :(

Reply:
done